hibernate-core/src/test/java/org/hibernate/test/locking/OptimisticLockTest.java

```diff
@@ -0,0 +1,92 @@
+package org.hibernate.test.locking;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.LockModeType;
+import javax.persistence.Version;
+
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.cfg.Configuration;
+
+import org.hibernate.testing.junit4.BaseCoreFunctionalTestCase;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class OptimisticLockTest extends BaseCoreFunctionalTestCase {
+
+	@Override
+	protected void configure(Configuration configuration) {
+		configuration.setProperty( AvailableSettings.GLOBALLY_QUOTED_IDENTIFIERS, "true" );
+	}
+
+	@Override
+	protected Class<?>[] getAnnotatedClasses() {
+		return new Class[] { Person.class };
+	}
+
+	@Before
+	public void setUp() {
+		inTransaction(
+				session -> {
+					Person person = new Person( "1", "Fabiana" );
+					session.persist( person );
+				}
+		);
+	}
+
+	@After
+	public void tearDown() {
+		inTransaction(
+				session -> {
+					session.createQuery( "delete from Person" ).executeUpdate();
+				}
+		);
+	}
+
+	@Test
+	public void testHqlQueryWithOptimisticLock() {
+		inTransaction(
+				session -> {
+					session.createQuery( "from Person e", Person.class )
+							.setLockMode( LockModeType.OPTIMISTIC )
+							.getResultList().get( 0 );
+				}
+		);
+	}
+
+	@Entity(name = "Person")
+	public static class Person {
+		@Id
+		private String id;
+
+		@Version
+		private long version1;
```

Comment:
Could we use 'version' directly if `GLOBALLY_QUOTED_IDENTIFIERS` has been set to true?